pymeasure/instruments/keithley/keithley2182.py

```diff
@@ -0,0 +1,399 @@
+        ":SENS:TEMP:RTEM?",
+        """Measure the internal temperature in Celsius."""
+    )
+
+    ##############
+    # Statistics #
+    ##############
+
+    buffer_points = Instrument.control(
+        ":TRAC:POIN?", ":TRAC:POIN %d",
+        """Control the number of buffer points.
+        This does not represent actual points in the buffer, but the configuration
+        value instead. Default is 100.""",
+        validator=truncated_range,
+        values=[2, 1024],
+        cast=int
+    )
+    mean = Instrument.measurement(
+        ":CALC2:FORM MEAN;:CALC2:STAT ON;:CALC2:IMM?;",
+        """Get the calculated mean (average) from the buffer data."""
+    )
+    maximum = Instrument.measurement(
+        ":CALC2:FORM MAX;:CALC2:STAT ON;:CALC2:IMM?;",
+        """Get the calculated maximum from the buffer data."""
+    )
+    minimum = Instrument.measurement(
+        ":CALC2:FORM MIN;:CALC2:STAT ON;:CALC2:IMM?;",
+        """Get the calculated minimum from the buffer data."""
+    )
+    standard_dev = Instrument.measurement(
+        ":CALC2:FORM SDEV;:CALC2:STAT ON;:CALC2:IMM?;",
+        """Get the calculated standard deviation from the buffer data."""
+    )
+
+    ###########
+    # Trigger #
+    ###########
+
+    trigger_count = Instrument.control(
+        ":TRIG:COUN?", ":TRIG:COUN %d",
+        """Control the trigger count which can take values from 1 to 9,999.
+        Default is 1.""",
+        validator=truncated_range,
+        values=[1, 9999],
+        cast=int
+    )
+    trigger_delay = Instrument.control(
+        ":TRIG:DEL?", ":TRIG:DEL %g",
+        """Control the trigger delay in seconds, which can take values from 0 to
+        999999.999 s. Default is 0.""",
+        validator=truncated_range,
+        values=[0, 999999.999]
+    )
+
+    ###########
+    # Methods #
+    ###########
+
+    def auto_line_frequency(self):
+        """Set appropriate limits for NPLC voltage and temperature readings."""
+        if self.line_frequency == 50:
+            self.temperature_nplc_values = [0.01, 50]
+            self.voltage_nplc_values = [0.01, 50]
+        else:
+            self.temperature_nplc_values = [0.01, 60]
+            self.voltage_nplc_values = [0.01, 60]
+
+    @property
+    def error(self):
+        """Get a tuple of an error code and message from a single error."""
+        err = self.values(":system:error?")
+        if len(err) < 2:
+            err = self.read()  # Try reading again
+        code = err[0]
+        message = err[1].replace('"', '')
+        return (code, message)
+
+    def check_errors(self):
+        """Log any system errors reported by the instrument."""
+        code, message = self.error
+        while code != 0:
+            t = time.time()
+            log.info("Keithley 2182 reported error: %d, %s", code, message)
+            code, message = self.error
+            if (time.time() - t) > 10:
+                log.warning("Timed out for Keithley 2182 error retrieval.")
```

Comment:
Check, whether that code is necessary, or it is sufficient to use the inherited method (from `Instrument`).